### PR TITLE
[jest-expo] create __REACT_DEVTOOLS_GLOBAL_HOOK__ stub

### PR DESCRIPTION
**Pull request by marklawlor**
# Why

Fix the React Navigation errors in 0.74

The issue is caused by this commit https://github.com/facebook/react-native/commit/a286f00073aa10b13bf5d78fae7701c1678108be

Here is the chain of code that causes the issue

1. React Native now checks for the existence of `__REACT_DEVTOOLS_GLOBAL_HOOK__` on startup if the `getInspectorDataForViewAtPoint` function is imported.
2. `getInspectorDataForViewAtPoint` is imported by `AppContainer` in its `DevtoolsOverlay` https://github.com/hoxyq/react-native/blob/fbbb4246707d85b692c006e0cb3b186a7c9068bc/packages/react-native/Libraries/ReactNative/AppContainer.js#L55
3. React Navigation `<DebugContainer />` creates a `<AppContainer />` in non-production environments https://github.com/react-navigation/react-navigation/blob/523fa4f3afb7c428a8370bfd83933c0163224e66/packages/native-stack/src/views/DebugContainer.native.tsx#L21-L35
4. `@react-navigation/native-stack` creates a `<DebugContainer />`

TLDR: Any project running `@react-navigation/native-stack` (the default Stack for Expo Router) in a non-production test environment will fail in 0.74

@tsapeta This PR only fixes this issue. The router tests in 0.74 are still failing due to `react-navigate-gesture-handler` errors in fabric

# How

Add a stub for `__REACT_DEVTOOLS_GLOBAL_HOOK__` in the `jest-expo` environment setup

# Test Plan

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/jest-expo/src/preset/setup.js [hunk 1]
```diff
@@ -18,6 +18,21 @@ if (typeof window !== 'object') {
   globalThis.window.navigator = {};
 }
 
+if (typeof globalThis.__REACT_DEVTOOLS_GLOBAL_HOOK__ === 'undefined') {
+  // RN 0.74 checks for the __REACT_DEVTOOLS_GLOBAL_HOOK__ on startup if getInspectorDataForViewAtPoint is used
+  // React Navigation uses getInspectorDataForViewAtPoint() for improved log box integration in non-production builds
+  globalThis.__REACT_DEVTOOLS_GLOBAL_HOOK__ = {
+    isDisabled: true, // Used by `react-test-renderer` https://github.com/facebook/react/blob/113ab9af08c46e8a548a397154f5c9dfeb96ab6a/packages/react-reconciler/src/ReactFiberDevToolsHook.js#L60
+    renderers: {
+      // https://github.com/hoxyq/react-native/blob/fbbb4246707d85b692c006e0cb3b186a7c9068bc/packages/react-native/Libraries/Inspector/getInspectorDataForViewAtPoint.js#L40
```
Comment (tsapeta) [inline packages/jest-expo/src/preset/setup.js:27]: Probably don't want to refer to a fork 😅

File: packages/jest-expo/src/preset/setup.js [hunk 2]
```diff
@@ -18,6 +18,21 @@ if (typeof window !== 'object') {
...
+      values: () => [],
+    },
+    on() {}, // https://github.com/hoxyq/react-native/blob/fbbb4246707d85b692c006e0cb3b186a7c9068bc/packages/react-native/Libraries/Inspector/getInspectorDataForViewAtPoint.js#L45
```
Comment (tsapeta) [inline packages/jest-expo/src/preset/setup.js:30]: Same here, it links to a fork

**Review by tsapeta — APPROVED** [review]
Looks good to me, thanks for your help! Could you add a changelog entry to `jest-expo` as well?
